### ISSUE TITLE
Offer friendly overloads that retain typedef struct handles

### ISSUE THREAD

**Issue by jnm2** [microsoft/CsWin32#150]
It looks like the HICON overload might have showed up as a followup to https://github.com/microsoft/CsWin32/issues/125#issuecomment-781058118:

> We may offer non-SafeHandle overloads as well, since that sounds like something that will help in some cases.

This overload saves me from the boilerplate of having to declare my own SafeHandle type to wrap my unowned handle value I got from `GetCursorInfo`, if I use that overload, now I have to add extra `unsafe` boilerplate instead of using `out var` for the second parameter.

Could there be a non-SafeHandle overload with a second `out` parameter?

**Comment by AArnott** [comment]
I totally get why you're asking for it, but every overload we add compounds the risk that C# will not be able to resolve which overload to invoke. In the `dotnet/pinvoke` project, we had 3-5 overloads per method, and it became a real problem where simple APIs required explicit type casts for `null` args just so the code would compile. Since adding overloads is typically less disruptive than removing them, we want to err on the side of adding them slowly and seeing where the worst pain is so we can add only the most impactful (for good!) overloads and omit the less useful ones. 

So we can keep this issue active to represent your vote for a particular set of overloads to be offered, but we are deliberately delaying "fixing" this till we get more feedback. I hope you'll understand. 😀

**Comment by AArnott** [comment]
> Is there any other concern in general with adding friendly overloads in this current phase of the project?

Perhaps not. But we may want to make other changes to a typedef handle overload.
Also: we have a couple of very large changes coming up (e.g. COM *interfaces*) so I'd like to delay any more work on friendly overloads till that work is done.

**Comment by jnm2** [comment]
@AArnott The commit at closed this doesn't look related at first glance. Was closing this issue a mistake?

**Comment by AArnott** [comment]
Yup. That's what comes of merging commits in from another repo with comments that mention fixing issues in those other repos. Thanks for catching.
